ipod/ipod.py

```diff
@@ -147,20 +149,36 @@ def ipod(
             solution=orbit_members_iter.solution,
         )
 
-        initial_reduced_chi2 = orbit_iter.reduced_chi2[0].as_py()
-        if initial_reduced_chi2 > rchi2_threshold:
-            # This can occur if the orbit was previously accepted with these
-            # observations at a higher reduced chi2 threshold than the one
-            # given to this function. In this case we will force a fit
-            # with the given observations to give this orbit a chance to improve.
+        # Now fit the orbit with the given observations so we make sure we have
+        # an accurate orbit to start with
+        orbit_iter_fit, orbit_members_iter_fit = od(
+            orbit_iter,
+            orbit_observations,
+            rchi2_threshold=rchi2_threshold,
+            min_obs=6,
+            min_arc_length=1.0,
+            contamination_percentage=0.0,
+            delta=1e-8,
+            max_iter=5,
+            method="central",
+            propagator=propagator,
+            propagator_kwargs=propagator_kwargs,
+        )
+        if len(orbit_iter_fit) == 0:
             logger.debug(
-                f"Initial reduced chi2 of {initial_reduced_chi2} "
-                f"is greater than the threshold of {rchi2_threshold}."
+                "Initial orbit fit with provided observations failed. "
```

Comment:
Are you finding this happens much?

Reply:
I don't have solid numbers but it does occur when $3 < \chi_\nu^2 \leq 10$ orbits from THOR are unable to improve below the lower threshold in this function.